sqldb/sqlc/migrations/000005_migration_tracker.up.sql

```diff
@@ -0,0 +1,19 @@
+-- migration_tracker is a table that keeps track of custom migrations that have
+-- been run on the database. This table is used to ensure that migrations are
+-- idempotent and are only run once.
+CREATE TABLE IF NOT EXISTS migration_tracker (
+        -- migration_id is the id of the migration.
+        migration_id TEXT NOT NULL,
+
+        -- migration_time is the timestamp at which the migration was run.
+        migration_time TIMESTAMP,
+
+        PRIMARY KEY (migration_id)
+);
+
+-- Insert the expected migrations to the migration tracker.
+INSERT INTO migration_tracker (
+    migration_id,
+    migration_time
+) VALUES 
```

Comment:
> I agree that this schema alone isn’t sufficient for managing dependencies between migrations. However, since we don’t currently have migrations with dependencies, I think keeping it simple for now is the best approach. wdyt?

Just trying to make sure things dont break on the next migration. so just want to know what the flow is if:
1) user is currently on kvdb_sql.
2) we merge this
3) we add a new Sql `up` file for a migration _on the invoices table_ (call it `up` file 7)
4) user upgrades their LND node at this point only (ie, not between steps 2 & 3)

Just want to make sure we are covered for this case. Cause if this happens, we want the current SQL migrations (1-6) to run, then we'd want this migration to run, and then we'd want up file 7 to run. Right? because this migration as it stands today depends on the DB being the in state as of up file 6. Or will we just keep adjusting this migration each time to account for the latest up file version?

Reply:
>  The timestamp itself should remain unchanged; it indicates that the migration has successfully occurred.

cool yeah sounds good

Reply:
if we agree that we dont want to have to keep adjusting migration code after writing it to keep adjusting for the latest _live_ state of the DB, then we should also consider copying the pattern we have used for all our `kvdb` migrations where we have a `migration` folder per migration and then duplicate all our codec code so that our queries at migration time remain stable.

Reply:
and then we need to find a way gain more control over exactly which migrations are run when (sqlc vs in-code migrations)

Reply:
what's also nice then is that we then dont have to add any migration specific methods to our main DB interface. This will keep usage strict & prevent accidental use of calls made only for migrations

Reply:
really nice catch @ellemouton, so we are going to do the code-level as in kvdb style with its own migration dir ?

Reply:
I think so yes